### PR TITLE
Added more Documentation.

### PR DESCRIPTION
**Pull request by zorg93**
Moved Screen Fade properties and methods from game to Screen class

### CODE REVIEW

**Comment by zorg93** [comment]
Tested all new vehicles on 1.0.757.2 and the spawned ok :)

File: source/scripting/Game.cs [hunk 1]
```diff
@@ -338,137 +390,253 @@ public static InputMode CurrentInputMode
...
-				if (chars[i] >= 'A' && chars[i] <= 'Z')
-				{
-					chars[i] = (char)(chars[i] + 32);
-				}
-			}
-
-			byte[] bytes = Encoding.UTF8.GetBytes(chars);
+			uint hash = 0;	  
+			
+			byte[] bytes = Encoding.ASCII.GetBytes(input.ToLower());
```
Comment (kagikn) [inline source/scripting/Game.cs]: I confirmed ```GET_HASH_KEY``` takes a string value in UTF8 and convert ASCII uppercase to lowercase, so we should revert this.

Comment (zorg93) [inline source/scripting/Game.cs]: Just looking over it again, it actually uses both. from what i can see if the string starts with a quote it ignores the quote, then uses ASCII uppercase to lowercase, then hashes until end of string or another quote is found.
If string doesnt start with a quote then it uses UTF8 combined with ASCII uppercase to lowercase on the whole string.

Comment (kagikn) [inline source/scripting/Game.cs]: The results can be different since ```ASCII.GetBytes``` replaces non-ASCII characters to replacement string.

Comment (zorg93) [inline source/scripting/Game.cs]: I reckon I may just use the ingame function

**Comment by zorg93** [comment]
@kagikn there we go, no issues with it now :)

**Pull request merged**